modules/scatteredBidAdapter.js

```diff
@@ -0,0 +1,175 @@
+        return { w, h };
+      });
+
+      imp.banner = {
+        format
+      };
+    };
+    return imp;
+  });
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  // 1.
+  isBidRequestValid: function (bid) {
+    const bidderDomain = deepAccess(bid, 'params.bidderDomain')
+    if (bidderDomain === undefined || bidderDomain === '') {
+      return false
+    }
+
+    const sizes = deepAccess(bid, 'mediaTypes.banner.sizes')
+    if (sizes === undefined || sizes.length < 1) {
+      return false
+    }
+
+    return true
+  },
+
+  // 2.
+  buildRequests: function (validBidRequests, bidderRequest) {
+    const ortb2Obj = bidderRequest.ortb2 || {};
+    let { user } = ortb2Obj
+
+    let site = getConfig('site') || {};
+    if (ortb2Obj.site) {
+      mergeDeep(site, ortb2Obj.site)
+    }
+    // site.page    URL of the page where the impression will be shown.
+    // site.domain  Domain of the site (e.g., “mysite.foo.com”).
+    if (!site.page && bidderRequest.refererInfo) {
+      site.page = bidderRequest.refererInfo.referer;
+      site.domain = parseUrl(bidderRequest.refererInfo.referer).hostname
+    }
+
+    const bidderDomain = getKeyOnAny(validBidRequests, 'params.bidderDomain')
+
+    const device = getConfig('device') || {};
+    device.w = device.w || window.innerWidth;
+    device.h = device.h || window.innerHeight;
+    device.ua = device.ua || navigator.userAgent;
+
+    const imps = buildImps(validBidRequests)
+
+    const tid = getKeyOnAny(validBidRequests, 'transactionId');
+
+    const request = {
+      id: bidderRequest.auctionId,
+      imp: imps,
+      site,
+      user,
+      device,
+      source: { tid }
+    };
+
+    if (bidderRequest && bidderRequest.gdprConsent) {
+      let { gdprApplies, consentString } = bidderRequest.gdprConsent;
+      deepSetValue(request, 'user.ext.consent', consentString);
+      deepSetValue(request, 'regs.ext.gdpr', gdprApplies ? 1 : 0);
+    }
+
+    const test = getKeyOnAny(validBidRequests, 'params.test') || 0
+    if (test) {
+      request.test = 1;
+    }
+
+    return {
+      method: 'POST',
+      url: 'https://' + bidderDomain,
+      data: JSON.stringify(request),
+      options: {
+        contentType: 'application/json'
+      },
+      bidRequests: validBidRequests
+    };
+  },
+
+  // 3.
+  interpretResponse: function (serverResponse, request) {
+    if (!serverResponse.body) {
+      return;
+    }
+
+    const { bidRequests } = request;
+    const { seatbid, cur } = serverResponse.body;
+    let idx = 0;
+    let results = []
+
+    if (seatbid !== undefined) {
+      for (let seat of seatbid) {
+        for (let bidResponse of seat.bid) {
+          const bidRequest = bidRequests[idx]
+
+          const result = {
+            requestId: bidRequest.bidId,
+            cpm: bidResponse.price,
+            creativeId: bidResponse.crid,
+            ttl: 360,
+            netRevenue: bidRequest.netRevenue === 'net',
```

Comment:
I think `result.netRevenue` would always be false in practice - since I don't know of any situation where `bidRequest.netRevenue` would be anything other than undefined. Could you explain the reasoning - who are you expecting to fill it out?